vnext/Microsoft.ReactNative/Modules/AccessibilityInfoModule.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "AccessibilityInfoModule.h"
+#include <UI.Xaml.Automation.Peers.h>
+#include <UI.Xaml.Controls.h>
+#include <XamlUtils.h>
+#include <winrt/Windows.ApplicationModel.DataTransfer.h>
+#include <winrt/Windows.UI.ViewManagement.h>
+#include "Unicode.h"
+#include "Utils/Helpers.h"
+
+namespace Microsoft::ReactNative {
+
+void AccessibilityInfo::Initialize(winrt::Microsoft::ReactNative::ReactContext const &reactContext) noexcept {
+  m_context = reactContext;
+}
+
+void AccessibilityInfo::isReduceMotionEnabled(std::function<void(React::JSValue const &)> const &onSuccess) noexcept {
+  auto jsDispatcher = m_context.JSDispatcher();
+  m_context.UIDispatcher().Post([weakThis = weak_from_this(), jsDispatcher, onSuccess] {
+    if (auto strongThis = weakThis.lock()) {
+      winrt::Windows::UI::ViewManagement::UISettings uiSettings;
+      auto animationsEnabled = uiSettings.AnimationsEnabled();
+      jsDispatcher.Post([animationsEnabled, onSuccess] { onSuccess(!animationsEnabled); });
+    }
+  });
+}
+
+void AccessibilityInfo::isTouchExplorationEnabled(
+    std::function<void(React::JSValue const &)> const &onSuccess) noexcept {
+  onSuccess(false);
```

Comment:
>false [](start = 12, length = 5)

Re:  - `isTouchExplorationEnabled` always returns false.  As far as I know windows does not expose a hook to know if someone is running narrator or similar tools.  Open question, should we default to true instead of false?

Idk what "touch exploration" means & how it's used, but as far as telling whether an a11y tool is running goes: There's UiaClientsAreListening ([https://docs.microsoft.com/en-us/windows/win32/api/uiautomationcoreapi/nf-uiautomationcoreapi-uiaclientsarelistening](https://docs.microsoft.com/en-us/windows/win32/api/uiautomationcoreapi/nf-uiautomationcoreapi-uiaclientsarelistening)). This is about events which is a subset of UIA functionality, but I'd assume that any "real" a11y tool would do that. If you should decide to use UiaClientsAreListening, do a sanity check on its returns; I vaguely recall that on earlier Win10 versions this returned true even if there was no real a11y tool running.

Reply:
Is that API safe for UWP usage?

Reply:
There's also IRawElementProviderAdviseEvents ([https://docs.microsoft.com/en-us/windows/win32/api/uiautomationcore/nn-uiautomationcore-irawelementprovideradviseevents](https://docs.microsoft.com/en-us/windows/win32/api/uiautomationcore/nn-uiautomationcore-irawelementprovideradviseevents)). This is something that needs to be implemented in the UI framework, and we'd have to connect to that - probably too complicated for what we want to achieve here.

---
In reply to: [478625704](https://github.com/microsoft/react-native-windows/pull/5849#discussion_r478625704) [](ancestors = 478625704)

Reply:
I believe so (we're using it in WinRT platform adaptors in FastAcc). Forrest would know for sure.

---
In reply to: [478626922](https://github.com/microsoft/react-native-windows/pull/5849#discussion_r478626922) [](ancestors = 478626922)

Reply:
Per otools/bin/mox/mosdk-shaped-fullanno.xml, it looks like UiaClientsAreListening is allowed.

---
In reply to: [478626922](https://github.com/microsoft/react-native-windows/pull/5849#discussion_r478626922) [](ancestors = 478626922)

Reply:
It still seems to return true even if nothing is running.  Not sure how useful that hook up is then.

Reply:
in-box XAML uses UiaClientsAreListening for this purpose, and I am pretty sure this is working reliably there (ie it returns true only when an AT like Narrator is running).

---
In reply to: [478640914](https://github.com/microsoft/react-native-windows/pull/5849#discussion_r478640914) [](ancestors = 478640914,478626922)

Reply:
@kmelmon it returns true if there is a UIA client that is subscribed to events - but UIA clients don't have to do that, instead they can just poll properties every N seconds. Narrator does subscribe to UIA events, so if the intent is to only get this working with narrator and maybe not support other UIA clients then that API is fine.

Reply:
It's been a long time since I looked at this API, but IIRC it could return false positives. For example, I *think* it returns true on touch devices even if there no ATs listening.